app/factories/bulkrax/object_factory.rb

```diff
@@ -0,0 +1,214 @@
+      return match if match
+    end
+
+    # An ActiveFedora bug when there are many habtm <-> has_many associations means they won't all get saved.
+    # https://github.com/projecthydra/active_fedora/issues/874
+    # 2+ years later, still open!
+    def create
+      attrs = transform_attributes
+      @object = klass.new
+      object.reindex_extent = Hyrax::Adapters::NestingIndexAdapter::LIMITED_REINDEX if object.respond_to?(:reindex_extent)
+      run_callbacks :save do
+        run_callbacks :create do
+          if klass == Collection
+            create_collection(attrs)
+          elsif klass == FileSet
+            create_file_set(attrs)
+          else
+            work_actor.create(environment(attrs))
+          end
+        end
+      end
+      object.apply_depositor_metadata(@user) && object.save! if object.depositor.nil?
+      log_created(object)
+    end
+
+    def log_created(obj)
+      msg = "Created #{klass.model_name.human} #{obj.id}"
+      Rails.logger.info("#{msg} (#{Array(attributes[work_identifier]).first})")
+    end
+
+    def log_updated(obj)
+      msg = "Updated #{klass.model_name.human} #{obj.id}"
+      Rails.logger.info("#{msg} (#{Array(attributes[work_identifier]).first})")
+    end
+
+    def log_deleted_fs(obj)
+      msg = "Deleted All Files from #{obj.id}"
+      Rails.logger.info("#{msg} (#{Array(attributes[work_identifier]).first})")
+    end
+
+    private
+
+    # @param [Hash] attrs the attributes to put in the environment
+    # @return [Hyrax::Actors::Environment]
+    def environment(attrs)
+      Hyrax::Actors::Environment.new(object, Ability.new(@user), attrs)
+    end
+
+    def work_actor
+      Hyrax::CurationConcern.actor
+    end
+
+    def create_collection(attrs)
+      attrs = clean_attrs(attrs)
+      attrs = collection_type(attrs)
+      object.attributes = attrs
+      object.save!
+    end
+
+    def update_collection(attrs)
+      object.attributes = attrs
+      object.save!
+    end
+
+    # This method is heavily inspired by Hyrax's AttachFilesToWorkJob
+    def create_file_set(attrs)
+      _, work = find_record(attributes[related_parents_parsed_mapping].first, importer_run_id)
+      work_permissions = work.permissions.map(&:to_hash)
+      attrs = clean_attrs(attrs)
+      file_set_attrs = attrs.slice(*object.attributes.keys)
+      object.assign_attributes(file_set_attrs)
+
+      attrs['uploaded_files'].each do |uploaded_file_id|
+        uploaded_file = ::Hyrax::UploadedFile.find(uploaded_file_id)
+        next if uploaded_file.file_set_uri.present?
+
+        actor = ::Hyrax::Actors::FileSetActor.new(object, @user)
+        uploaded_file.update(file_set_uri: actor.file_set.uri)
+        actor.file_set.permissions_attributes = work_permissions
+        actor.create_metadata
+        actor.create_content(uploaded_file)
+        actor.attach_to_work(work)
+      end
+
+      object.save!
+    end
+
```

Comment:
Was `create_file_set_actor` removed here? If so, can you add a comment stating that it was removed along with the reason why?